fabric-client-tags-api-v1/src/client/java/net/fabricmc/fabric/api/tag/client/v1/ClientTags.java

```diff
@@ -126,16 +100,27 @@ public static <T> boolean isInWithLocalFallback(TagKey<T> tagKey, RegistryEntry<
 		Optional<? extends Registry<T>> maybeRegistry = getRegistry(tagKey);
 
 		if (maybeRegistry.isPresent()) {
+			// Check the synced tag exists and use that
 			if (maybeRegistry.get().getEntryList(tagKey).isPresent()) {
 				return registryEntry.isIn(tagKey);
 			}
 		}
 
-		if (registryEntry.getKey().isPresent()) {
-			return isInLocal(tagKey, registryEntry.getKey().get());
+		if (registryEntry.getKey().isEmpty()) {
+			// No key?
+			return false;
 		}
 
-		return false;
+		// Recursively search the entries contained with the tag
+		ClientTagsLoader.LoadedTag wt = getOrCreatePartiallySyncedTag(tagKey);
+		boolean isIn = wt.immediateChildIds().contains(registryEntry.getKey().get().getValue());
```

Comment:
Nitpick: this variable isn't useful, it could be replaced by returning true and false directly in the right places